File: 3_gemini_sol2.cpp
#include <iostream>

using namespace std;

long long calculateGreatestCommonDivisor(long long a, long long b) {
    while (b != 0) {
        long long remainder = a % b;
        a = b;
        b = remainder;
    }
    return a;
}

int main() {
    long long modulus;
    if (cin >> modulus) {
        long long remainingValue = modulus;
        long long squareFreeProduct = 1;

        for (long long divisor = 2; divisor * divisor <= remainingValue; ++divisor) {
            if (remainingValue % divisor == 0) {
                squareFreeProduct *= divisor;
                while (remainingValue % divisor == 0) {
                    remainingValue /= divisor;
                }
            }
        }
        
        if (remainingValue > 1) {
            squareFreeProduct *= remainingValue;
        }

        long long cursedPairCount = 0;
        for (long long startX = 1; startX < modulus; ++startX) {
            long long currentGcd = calculateGreatestCommonDivisor(startX, squareFreeProduct);
            long long requiredYFactor = squareFreeProduct / currentGcd;
            cursedPairCount += (modulus - 1) / requiredYFactor;
        }

        cout << cursedPairCount << "\n";
    }
    return 0;
}

File: 3_gemini_sol2.java
import java.util.Scanner;

public class SequenceSolver {
    private static long calculateGreatestCommonDivisor(long first, long second) {
        while (second != 0) {
            long remainder = first % second;
            first = second;
            second = remainder;
        }
        return first;
    }

    public static void main(String[] args) {
        Scanner inputScanner = new Scanner(System.in);
        if (inputScanner.hasNextLong()) {
            long modulus = inputScanner.nextLong();
            long remainingValue = modulus;
            long squareFreeProduct = 1;
            
            for (long divisor = 2; divisor * divisor <= remainingValue; divisor++) {
                if (remainingValue % divisor == 0) {
                    squareFreeProduct *= divisor;
                    while (remainingValue % divisor == 0) {
                        remainingValue /= divisor;
                    }
                }
            }
            
            if (remainingValue > 1) {
                squareFreeProduct *= remainingValue;
            }

            long cursedPairCount = 0;
            for (long startX = 1; startX < modulus; startX++) {
                long currentGcd = calculateGreatestCommonDivisor(startX, squareFreeProduct);
                long requiredYFactor = squareFreeProduct / currentGcd;
                cursedPairCount += (modulus - 1) / requiredYFactor;
            }
            
            System.out.println(cursedPairCount);
        }
        inputScanner.close();
    }
}
